File: python/determine-the-minimum-sum-of-a-k-avoiding-array.py
class Solution:
    def minimumSum(self, n: int, k: int) -> int:
        output = 0
        current = 0
        for current in range(1, min(n + 1, k // 2 + 1)):
            output += current
        for current in range(k, k + n - current):
            output += current
        return output

File: go/determine-the-minimum-sum-of-a-k-avoiding-array.go
func minimumSum(n int, k int) int {
	output := 0
	current := 0
	firstLimit := (k / 2) + 1
	if (n + 1) < firstLimit {
		firstLimit = n + 1
	}
	for current < firstLimit {
		output += current
		current += 1
	}
	secondLimit := k + n - current
	current = k
	for current <= secondLimit {
		output += current
		current += 1
	}
	return output
}
